config/webpack/environment.js

```diff
@@ -1,11 +1,27 @@
 const { environment } = require('@rails/webpacker')
 
-  const webpack = require('webpack')
-  environment.plugins.prepend('Provide',
-    new webpack.ProvidePlugin({
-      $: 'jquery/src/jquery',
-      jQuery: 'jquery/src/jquery'
-    })
-  )
-
-module.exports = environment
+const webpack = require('webpack')
+
+// Add an ProvidePlugin
+environment.plugins.prepend('Provide',  new webpack.ProvidePlugin({
+    $: 'jquery',
+    jQuery: 'jquery',
+    jquery: 'jquery',
+    'window.Tether': "tether",
+    Popper: ['popper.js', 'default'],
+    ActionCable: 'actioncable',
+    Vue: 'vue',
+    VueResource: 'vue-resource',
+  })
+)
+
+const config = environment.toWebpackConfig()
+
+config.resolve.alias = {
+  jquery: "jquery/src/jquery",
+  vue: "vue/dist/vue.js",
+  vue_resource: "vue-resource/dist/vue-resource",
+}
+
+// export the updated config
+module.exports = environment
```

Comment:
Please add a newline to the end of the file :construction_worker_man:
```suggestion
module.exports = environment

```